DQMOffline/RecoB/plugins/MiniAODTaggerAnalyzer.cc

```diff
@@ -0,0 +1,85 @@
+#include "DQMOffline/RecoB/plugins/MiniAODTaggerAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+
+MiniAODTaggerAnalyzer::MiniAODTaggerAnalyzer(const edm::ParameterSet& pSet)
+    : jetToken_(consumes<std::vector<pat::Jet> >(pSet.getParameter<edm::InputTag>("JetTag"))),
+      disrParameters_(pSet.getParameter<edm::ParameterSet>("parameters")),
+
+      folder_(pSet.getParameter<std::string>("folder")),
+      discrNumerator_(pSet.getParameter<vstring>("numerator")),
+      discrDenominator_(pSet.getParameter<vstring>("denominator")),
+
+      mclevel_(pSet.getParameter<int>("MClevel")),
+      doCTagPlots_(pSet.getParameter<bool>("CTagPlots")),
+      dodifferentialPlots_(pSet.getParameter<bool>("differentialPlots")),
+      discrCut_(pSet.getParameter<double>("discrCut")),
+
+      etaActive_(pSet.getParameter<bool>("etaActive")),
+      etaMin_(pSet.getParameter<double>("etaMin")),
+      etaMax_(pSet.getParameter<double>("etaMax")),
+      ptActive_(pSet.getParameter<bool>("ptActive")),
+      ptMin_(pSet.getParameter<double>("ptMin")),
+      ptMax_(pSet.getParameter<double>("ptMax"))
+
+{}
+
+MiniAODTaggerAnalyzer::~MiniAODTaggerAnalyzer() {}
+
+void MiniAODTaggerAnalyzer::bookHistograms(DQMStore::IBooker& ibook, edm::Run const& run, edm::EventSetup const& es) {
+  jetTagPlotter_ = std::make_unique<JetTagPlotter>(folder_,
+                                                   EtaPtBin(etaActive_, etaMin_, etaMax_, ptActive_, ptMin_, ptMax_),
+                                                   disrParameters_,
+                                                   mclevel_,
+                                                   false,
+                                                   ibook,
+                                                   doCTagPlots_,
+                                                   dodifferentialPlots_,
+                                                   discrCut_);
+}
+
+void MiniAODTaggerAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<std::vector<pat::Jet> > jetCollection;
+  iEvent.getByToken(jetToken_, jetCollection);
+
+  // Loop over the pat::Jets
+  for (std::vector<pat::Jet>::const_iterator jet = jetCollection->begin(); jet != jetCollection->end(); ++jet) {
+    // fill numerator
+    float numerator = 0;
+    for (const auto& discrLabel : discrNumerator_) {
+      numerator = numerator + jet->bDiscriminator(discrLabel);
+    }
+
+    // fill denominator
+    float denominator;
+    if (discrDenominator_.empty()) {
+      denominator = 1;
```

Comment:
I still miss why do you want to add to histograms even when the list of discriminators is empty: is this really needed?
Shouldn't you more conveniently initialize `float denominator = 0;` and avoid adding to histo if there are no denominators?

Reply:
Yes, this is used for `BvsAll`. In this case the denominator is the sum of all probabilities.
So instead of summing up all probabilities to numerically 1 in the loop, the denominator is set to 1.

Reply:
Ok, now I understand, thank you.
Still, wouldn't it be more intutitive if in that case the first string in `discrDenominator_` is set as (e.g.) "ALL"? With empty lists one normally expects no elements insides, not "the sum of all probabilities"...

Reply:
This would be intuitive but requires a check for an empty list and a comparison between strings in each event. As the denominator list cannot be empty anyways I think it's a good way to interpret empty lists as no division performed.
I'll add a comment to explain this case.